src/roman_datamodels/random_utils.py

```diff
@@ -227,3 +227,38 @@ def generate_optical_element(seed=None):
 def generate_software_version(seed=None):
     vals = generate_positive_int(100, size=3, seed=seed)
     return f"{vals[0]}.{vals[1]}.{vals[2]}"
+
+
+def generate_read_pattern(seed=None):
+    rng = np.random.default_rng(seed)
+
+    length = rng.integers(16, 512, endpoint=True)
+
+    base_list = np.zeros(length, dtype=int)
+    choices = [True, False]
+
+    # Organize groups
+    group = 1
+    for index in range(length):
+        if rng.choice(choices):
+            group += 1
+
+        base_list[index] = group
+
+    # Randomly omit
+    for index in range(length):
+        if rng.choice(choices):
+            base_list[index] = 0
+
+    read_pattern = {}
+    for index in range(length):
+        group = base_list[index]
+        if group == 0:
+            continue
+
+        if group not in read_pattern:
+            read_pattern[group] = []
+
+        read_pattern[group].append(index)
+
+    return [read_pattern[key] for key in sorted(read_pattern.keys())]
```

Comment:
This is separate, but there was not a random `read_pattern` generator... So I wrote one in an early iteration of these changes. I think this should be a separate PR.
```suggestion
```